tests/test_detectAndMeasure.py

```diff
@@ -594,6 +597,35 @@ def test_mask_streaks(self):
         streakMaskSet = (outMask & streakMask) > 0
         self.assertTrue(np.all(streakMaskSet[20:23, 40:200]))
 
+        # Add a more trapezoid shaped streak across an image that is
+        # fainter and check that it is also detected
+        bbox = science.getBBox()
+        difference = science.clone()
+        difference.maskedImage -= matchedTemplate.maskedImage
+        width = 100
+        amplitude = 5
+        x0 = -100 + bbox.getBeginX()
+        y0 = -100 + bbox.getBeginY()
+        x1 = xSize + x0 + 100
+        y1 = ySize/2
+        corner_coords = [lsst.geom.Point2D(x0, y0),
+                         lsst.geom.Point2D(x0, y0 + width),
+                         lsst.geom.Point2D(x1, y1),
+                         lsst.geom.Point2D(x1 + width, y1)]
+        streak_trapezoid = afwGeom.Polygon(corner_coords)
+        streak_image = streak_trapezoid.createImage(bbox)
+        streak_image *= amplitude
+        difference.image.array += streak_image.array
+        output = detectionTask.run(science, matchedTemplate, difference)
+        outMask = output.subtractedMeasuredExposure.mask.array
+        streakMask = output.subtractedMeasuredExposure.mask.getPlaneBitMask("STREAK")
+        streakMaskSet = (outMask & streakMask) > 0
+        # Check that pixel values in streak_image equal the streak amplitude
+        streak_check = np.where(streak_image.array == amplitude)
+        self.assertTrue(np.all(streakMaskSet[streak_check]))
+        # Check that the entire image was not masked STREAK
+        self.assertFalse(np.all(streakMaskSet))
+
```

Comment:
If error handling is needed in the main code, then you should include a null test where things go wrong so you can check that the error is properly caught.

Reply:
Not relevant since we can never land here without a valid PSF